sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -547,6 +547,10 @@ valueExpression
     | left=valueExpression comparisonOperator right=valueExpression                          #comparison
     ;
 
+builtinOpExpression
+    : primaryExpression (CONCAT_PIPE primaryExpression)*                                       #concat
```

Comment:
This is interesting...So seems mysql and postgres has different precedence for it.